structured_data/2021_06_30_tensorflow_data_validation/tfdv_functionality_showcase.ipynb

```diff
@@ -0,0 +1,508 @@
+        "* Setting the minimum required presence of a feature: `feature.presence.min_fraction=0.9`\n",
+        "* Changing the valency of a feature: `feature.value_count.min = min` or `feature.value_count.max = max`"
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "Djwii2jZVpJL"
+      },
+      "source": [
+        "# Environment-based schema\n",
+        "\n",
+        "The schema can differ per environment. For example: "
+      ]
+    },
+    {
+      "cell_type": "code",
+      "metadata": {
+        "id": "blVQeHjhTVbw"
+      },
+      "source": [
+        "serving_stats = tfdv.generate_statistics_from_dataframe(serving_data)\n",
+        "serving_anomalies = tfdv.validate_statistics(serving_stats, schema)\n",
+        "\n",
+        "tfdv.display_anomalies(serving_anomalies)"
+      ],
+      "execution_count": null,
+      "outputs": []
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "8QjBUkws9Tj4"
+      },
+      "source": [
+        "Here we see that the tips value is missing from the serving data. This is due to the fact that this is our label value, which is absent from serving data.\n",
+        "\n",
+        "This is were environments come in. You can create extra environments for your schema using `schema.default_environment.append('env_name')` and add or remove a feature from this using `feature.in_environment.append('env_name')` or `feature.not_in_environment.append('env_name')`"
+      ]
+    },
+    {
+      "cell_type": "code",
+      "metadata": {
+        "id": "dZlNWX3ISxVP"
+      },
+      "source": [
+        "schema.default_environment.append('TRAINING')\n",
+        "schema.default_environment.append('SERVING')\n",
+        "\n",
+        "tips_feature = tfdv.get_feature(schema, 'tips')\n",
+        "tips_feature.not_in_environment.append('SERVING')\n",
+        "\n",
+        "serving_anomalies = tfdv.validate_statistics(serving_stats, schema, environment='SERVING')\n",
+        "tfdv.display_anomalies(serving_anomalies)"
+      ],
+      "execution_count": null,
+      "outputs": []
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "71DAstmGbicI"
+      },
+      "source": [
+        "# Distribution Drift & Skew"
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "8upbZbX7gOU6"
+      },
+      "source": [
+        "Using the skew & drift comparator, you can configure your schema to allow detection of drift and skew. Both comparators are essentially the same. The skew comparator mean to be used to detect changes between training and serving statistics, and the drift comparator is used to detect changes between two different datasets in time.\n",
+        "\n",
+        "This drift and skew is measured using the following divergence metrics:\n",
```

Comment:
@julienschuermans Also added it in the notebook itself

Reply:
> The skew comparator mean to be used to detect changes

typo?

Reply:
yep, fixed!